examples/sw/simple_system/hello_test/hello_test.c

```diff
@@ -17,5 +20,9 @@ int main(int argc, char **argv) {
 
   pcount_enable(0);
 
+  // Wait for 2 time periods
+  while (get_elapsed_time() != 2)
+    ;
+
```

Comment:
Just to give some more output perhaps do a loop where it outputs something periodically for a few ticks?

```c

timer_enable(2000);

uint64_t last_elapsed_time = get_elapsed_time();

while (last_elapsed != 4) {
    uint64_t cur_time = get_elapsed_time();
    if(cur_time != last_elapsed_time) {
        last_elapsed_time = cur_time;

        if(last_elapsed_time & 1) {
            puts("Tick!\n");
        } else {
            puts("Tock\n");
        }
        
    }

    asm volatile ("wfi");
}
```

Suspect you'll need a reasonably long timer to ensure you see every tick as the puts will need some time to execute (and you'll want to move the timer enable directly above my proposed loop to avoid missing a tick or two).

Reply:
Added thanks - that's a better example (and it gives an example of wfi behavior too).